src/main/java/duke/Task.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+
+public abstract class Task {
+    protected String desc;
+    protected boolean isComplete;
+
+    public Task(boolean isComplete, String desc) throws DukeException.EmptyDescriptionException {
+        this.desc = desc.trim();
+        if (desc.trim().length() == 0) {
+            throw new DukeException.EmptyDescriptionException();
+        }
+        this.isComplete = isComplete;
+    }
+
+    public void markAsComplete() {
+        this.isComplete = true;
+    }
+
+    public String getRepr() {
+        return String.format("%d|%s", this.isComplete ? 1 : 0, this.desc);
+    };
+
+    public static Task fromRepr(String s) throws DukeException {
+        if (!s.matches("[TED][|][01][|].+[|].*")) {
+            throw new DukeException.ParseException(s);
+        }
+        String[] arr = s.split("[|]");
```

Comment:
Perhaps you can consider creating a private static final variable to hold this magic literal for the delimiter split as well? I notice this issue in other places as well.